### PR TITLE
Spark RSR docs for FilOz review

### PR DESCRIPTION
**Pull request by BigLep**
Putting Spark RSR docs in a PR so easier to collect questions/suggestions from FilOz during their 2024-11-10 review before Space Meridian meeting.

FilOz: please add suggestions, ask contextual comments, etc.

### CODE REVIEW

File: service-level-indicators/spark-retrieval-success-rate.md [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+    },
+    ...
+  ]
+}
+```
+
+- Created by X on a daily basis as part of Y
+- Consumed by anyone wanting to analyze or display SP retrievability performance.
+
+## Callouts/Concerns with this SLI
```
Comment (BigLep) [inline service-level-indicators/spark-retrieval-success-rate.md:333]: TODO: add one that checker traffic is pretty easy to differentiate from real traffic (given checker traffic only requests the `payloadCid`).  Basically a SP could just return to payloadCid requests, have 100% Spark RSR, but actually not enabling guaranteed retrieval.

File: service-level-indicators/spark-retrieval-success-rate.md [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+
+In the future, and when Spark is compatible with Direct Data Onboarding (DDO), there will be real-time deal ingestion into the Spark Eligible Deal database when storage deals are made ([GitHub tracking issue](https://github.com/space-meridian/roadmap/issues/144)). This will mean that new SPs will not need to wait for up to a week to get a Spark score.
+
+The end result of the Deal Ingestion step is a database of all Eligible Deals that should be retrievable.
+
+## Task Sampling
+
+Each round of the Spark protocol is approximately 20 minutes. At the start of each round, the [Spark tasking service randomly selects a set of records](https://github.com/filecoin-station/spark-api/blob/f77aa4269ab8c19ff64b9b9ff22462c29a6b8514/api/lib/round-tracker.js#L310) from all [Eligible Deal](#eligible-deal) . We refer to each of these records as a [**Retrieval Task**](#retrieval-task)  [](Spark%20Request-Based%20(Non-Committee)%20Global%20Retriev%204c5e8c47c45f467f80392d00cac2aae4.md)and, specifically, it is the combination of an Eligible Deal and a Round Id. We will also refer to the set of Retrieval Tasks in the round as the [Round Retrieval Task List](#round-retrieval-task-list).
```
Comment (BigLep) [inline service-level-indicators/spark-retrieval-success-rate.md:81]:
```suggestion
Each round of the Spark protocol is approximately 20 minutes. At the start of each round, the [Spark tasking service randomly selects a set of records](https://github.com/filecoin-station/spark-api/blob/f77aa4269ab8c19ff64b9b9ff22462c29a6b8514/api/lib/round-tracker.js#L310) from all [Eligible Deals](#eligible-deal). We refer to each of these records as a [**Retrieval Task**](#retrieval-task)  and, specifically, it is the combination of an Eligible Deal and a Round Id. We will also refer to the set of Retrieval Tasks in the round as the [Round Retrieval Task List](#round-retrieval-task-list). 
```

**Review by BigLep — COMMENTED** [review]
2024-11-10 conversation:
* PayloadCID 
  * Label field should have the CID of the CAR that is stored.  
  * Spark wants to treat "payload" as the data that a user wants to store.  
  * Note: need to make it really clear on what an SP should do.  Document the conventions.  
  * Really want to either
     * Make deals with CARv2
     * Deal should have the index of the data stored
* IPNI needs to do some sort of sticky cache to keep results for at least 20 minutes
* Is it needed in IPNI to get pieceCids to CIDs
* Spark needs to do more investigation on how to ensure spark checks are indiscerable from real requests.  Right now a smart SP can see what are the Spark checks and then just be on their best behavior for the expected spark checks.
* Masih's video on CommPact: https://www.youtube.com/watch?v=ndLKjXQ1awI&list=PL_0VrY55uV19y0b03a-CXNuvlgJR3G0EG&index=15
* Steven idea of doing IPLD checks and pieceCid/range checks
* 🎬 Schedule recurring Spark RSR working group

**Comment by juliangruber** [comment]
> Right now a smart SP can see what are the Spark checks and then just be on their best behavior for the expected spark checks.

How could a SP do that?

**Comment by BigLep** [comment]
> > Right now a smart SP can see what are the Spark checks and then just be on their best behavior for the expected spark checks.
> 
> How could a SP do that?

@juliangruber : my understanding is that an SP can look at the [Round Retrieval Task List](https://api.filspark.com/rounds/current), see which rows they are the listed "minerId" for, and then just make sure to serve the corresponding payloadCids in that 20 minute period.

File: service-level-indicators/spark-retrieval-success-rate.md (hunk 1 above)
Comment (BigLep) [reply service-level-indicators/spark-retrieval-success-rate.md:333]: Done in https://github.com/filecoin-project/service-classes/pull/7. Resolving.

File: service-level-indicators/spark-retrieval-success-rate.md (hunk 2 above)
Comment (BigLep) [reply service-level-indicators/spark-retrieval-success-rate.md:81]: Done in https://github.com/filecoin-project/service-classes/pull/7 .  Resolving.

File: service-level-indicators/spark-retrieval-success-rate.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bajtos) [reply service-level-indicators/spark-retrieval-success-rate.md:333]: Let's discuss the ideas in https://github.com/space-meridian/roadmap/issues/190

**Comment by BigLep** [comment]
I'm going to close this PR as this isn't expected to be merged and it served its purpose.

**Pull request closed without merging**